src/users/auth-token.entity.ts

```diff
@@ -6,20 +6,40 @@
 
 import {
   Column,
+  CreateDateColumn,
   Entity,
   ManyToOne,
   PrimaryGeneratedColumn,
-} from 'typeorm/index';
+} from 'typeorm';
 import { User } from './user.entity';
 
 @Entity()
 export class AuthToken {
   @PrimaryGeneratedColumn()
   id: number;
 
-  @ManyToOne((_) => User, (user) => user.authToken)
+  @ManyToOne((_) => User, (user) => user.authTokens)
   user: User;
 
   @Column()
+  identifier: string;
+
+  @CreateDateColumn()
+  createdAt: Date;
+
+  @Column({ unique: true })
   accessToken: string;
+
+  public static create(
+    user: User,
+    identifier: string,
+    accessToken: string,
```

Comment:
Good token implementations also provide a last-used field allowing better management of tokens. Another addition could be a TTL or expires-at field to ensure tokens don't survive by accident.

Reply:
I guess this would be rather easy to implement with JWTs.